src/engine/SpatialJoin.cpp

```diff
@@ -364,35 +350,40 @@ VariableToColumnMap SpatialJoin::computeVariableToColumnMap() const {
 
   if (!(childLeft_ || childRight_)) {
     // none of the children has been added
-    variableToColumnMap[leftChildVariable.value()] = makeCol(ColumnIndex{0});
-    variableToColumnMap[rightChildVariable.value()] = makeCol(ColumnIndex{1});
+    variableToColumnMap[leftChildVariable_.value()] = makeCol(ColumnIndex{0});
+    variableToColumnMap[rightChildVariable_.value()] = makeCol(ColumnIndex{1});
   } else if (childLeft_ && !childRight_) {
     // only the left child has been added
-    variableToColumnMap[rightChildVariable.value()] = makeCol(ColumnIndex{1});
+    variableToColumnMap[rightChildVariable_.value()] = makeCol(ColumnIndex{1});
   } else if (!childLeft_ && childRight_) {
     // only the right child has been added
-    variableToColumnMap[leftChildVariable.value()] = makeCol(ColumnIndex{0});
+    variableToColumnMap[leftChildVariable_.value()] = makeCol(ColumnIndex{0});
   } else {
     // both children have been added to the Operation
     auto varColsLeftMap = childLeft_->getVariableColumns();
     auto varColsRightMap = childRight_->getVariableColumns();
     auto varColsLeftVec = copySortedByColumnIndex(varColsLeftMap);
     auto varColsRightVec = copySortedByColumnIndex(varColsRightMap);
-    size_t index = 0;
-    if (addDistToResult) {
-      variableToColumnMap[Variable{nameDistanceInternal}] =
-          makeCol(ColumnIndex{index});
-      index++;
-    }
-    for (size_t i = 0; i < varColsLeftVec.size(); i++) {
-      variableToColumnMap[varColsLeftVec.at(i).first] =
-          makeCol(ColumnIndex{index});
-      index++;
-    }
-    for (size_t i = 0; i < varColsRightVec.size(); i++) {
-      variableToColumnMap[varColsRightVec.at(i).first] =
-          makeCol(ColumnIndex{index});
-      index++;
+    auto sizeLeft = childLeft_->getResultWidth();
+    auto sizeRight = childRight_->getResultWidth();
+    // in case a result table contains entries, which aren't contained
+    // in the varColMap
+    std::ranges::for_each(
+        varColsLeftVec,
+        [&](std::pair<Variable, ColumnIndexAndTypeInfo> varColEntry) {
+          variableToColumnMap[varColEntry.first] =
+              makeCol(ColumnIndex{varColEntry.second.columnIndex_});
```

Comment:
You shouldn't always return `possiblyUndefined` but exactly preserve the input status.
You can do this by just copying the `varColEntry.second`, overwriting the columnIndex, and then addint this to the map.